### PR TITLE
Implemented update_consensus_key support in Contract API

### LINKED ISSUE #2067: Support new Lima operation (Update consensus key) on the Contract API
Support the new operation `Update consensus key` on the contract and estimate providers

**Is your feature request related to a problem and use case? Please describe.**
Update_consensus_key is a new operation that updates the consensus key of the baker to public_key starting from the current cycle plus PRESERVED_CYCLES + 1

```
{
      "kind": "update_consensus_key",
      "source": "string",
      "fee": "string",
      "counter": "string",
      "gas_limit": "string",
      "storage_limit": "string",
      "pk": "string"
}
```

**Describe the solution you'd like**
- Add support for the operation in the `RpcContractProvider`, and `RPCEstimateProvider` classes (as well as to the associated interfaces)
- Add unit tests
- Add integration tests
- Add documentation on the website

**Additional context**
https://tezos.gitlab.io/protocols/015_lima.html#consensus-key
https://gitlab.com/tezos/tezos/-/blob/master/docs/api/lima-openapi.json#/default/post_helpers_forge_operations

### PR DESCRIPTION
**Pull request by dsawali**
closes #2067 

- Implemented update_consensus_key in Contract APi
- Updated unit and integration tests
- Removed gitignore that prevents husky precommit hook to run
- Updated types for update_consensus_key

### CODE REVIEW

File: .husky/_/husky.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
+#!/bin/sh
```
Comment (roxaneletourneau) [inline .husky/_/husky.sh:1]: Is this for the pre-commit hook?

File: integration-tests/contract-update-consensus-key-operation.spec.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  describe(`Test Update Consensus Key using: ${rpc}`, () => {
+    let pk: string;
+    beforeAll(async (done) => {
+      await setup(true);
+
+      try {
+        const account = await createAddress();
+        pk = await account.signer.publicKey();
+        const pkh = await account.signer.publicKeyHash();
+        console.log(pkh);
```
Comment (roxaneletourneau) [inline integration-tests/contract-update-consensus-key-operation.spec.ts:18]: These 2 console.log can be removed

File: packages/taquito/src/contract/interface.ts [hunk 3]
```diff
@@ -238,4 +240,14 @@ export interface ContractProvider extends StorageProvider {
...
    */
   proposals(params: ProposalsParams): Promise<ProposalsOperation>;
+
+  /**
+   *
+   * @description Update consensus key
+   *
+   * @returns An operation handle with the result from the RPC node
+   *
+   * @param UpdateConsensusKeyParams Proposals operation parameter
```
Comment (roxaneletourneau) [inline packages/taquito/src/contract/interface.ts:250]: Proposals? Seems to be a copy paste error

File: packages/taquito/src/operations/operation-emitter.ts [hunk 4]
```diff
@@ -201,6 +201,13 @@ export abstract class OperationEmitter {
               ...op,
               period: currentVotingPeriod?.voting_period.index,
             };
+          case OpKind.UPDATE_CONSENSUS_KEY:
+            return {
+              ...op,
+              ...getSource(op),
+              ...getFee(op),
+              pk: op.pk,
```
Comment (roxaneletourneau) [inline packages/taquito/src/operations/operation-emitter.ts:209]: I don't think this line (`pk: op.pk`) is needed. 
`case OpKind.UPDATE_CONSENSUS_KEY` could be moved above with 
```
case OpKind.REVEAL:
case OpKind.DELEGATION:
case OpKind.REGISTER_GLOBAL_CONSTANT:
case OpKind.TX_ROLLUP_ORIGINATION:
case OpKind.TX_ROLLUP_SUBMIT_BATCH:
```

File: packages/taquito/test/contract/rpc-contract-provider.spec.ts [hunk 5]
```diff
@@ -1790,6 +1792,103 @@ describe('RpcContractProvider test', () => {
...
+
+      expect(result.raw).toEqual({
+        opbytes: 'test',
+        opOb: {
+          branch: 'test',
+          contents: [
+            {
+              kind: 'update_consensus_key',
+              source: 'test_pub_key_hash',
+              fee: '475',
```
Comment (roxaneletourneau) [inline packages/taquito/test/contract/rpc-contract-provider.spec.ts:1813]: suggestion for fee, gas_limit and storage_limit properties, you could take the value from estimate :
```
fee: estimate.suggestedFeeMutez.toString(),
gas_limit: estimate.gasLimit.toString(),
storage_limit: estimate.storageLimit.toString(),
```

File: packages/taquito/test/contract/rpc-contract-provider.spec.ts [hunk 6]
```diff
@@ -1790,6 +1792,103 @@ describe('RpcContractProvider test', () => {
...
+      expect(result.raw).toEqual({
+        opbytes: 'test',
+        opOb: {
+          branch: 'test',
+          contents: [
+            revealOp('test_pub_key_hash'),
+            {
+              kind: 'update_consensus_key',
+              source: 'test_pub_key_hash',
+              fee: '475',
```
Comment (roxaneletourneau) [inline packages/taquito/test/contract/rpc-contract-provider.spec.ts:1844]: Same comment as above

File: .husky/_/husky.sh (hunk 1 above)
Comment (dsawali) [reply .husky/_/husky.sh:1]: yes, just a husky script for the pre-commit hook 😄

File: integration-tests/contract-update-consensus-key-operation.spec.ts (hunk 2 above)
Comment (dsawali) [reply integration-tests/contract-update-consensus-key-operation.spec.ts:18]: whoops, removing

File: packages/taquito/src/contract/interface.ts (hunk 3 above)
Comment (dsawali) [reply packages/taquito/src/contract/interface.ts:250]: will update 😄

**Review by roxaneletourneau — APPROVED** [review]

**Review by zainen — APPROVED** [review]

**Pull request merged**